tracing/span.go

```diff
@@ -66,6 +79,29 @@ func newSpan(tracer *Tracer, spanType SpanType) *Span {
 			ZipkinBinaryAnnotationKeyComponent: baseplateComponent,
 		},
 	}
+	return span
+}
+
+// CreateServerSpan creates a new Server Span, calls any registered
+// BaseplateHooks, and starts the Span.
+func CreateServerSpan(tracer *Tracer, name string) *Span {
+	span := newSpan(tracer, name, SpanTypeServer)
+	onServerSpanCreate(span)
+	span.startSpan()
+	return span
+}
+
+func (s *Span) startSpan() {
+	for _, hook := range s.hooks {
+		if err := hook.OnStart(s); err != nil {
+			log.Error("OnStart hook error: " + err.Error())
```

Comment:
nit: When using the zap logger, prefer to do this instead: `log.Errorw("OnStart hook error", "err", err)`. Or we can use the wrapped simple logger in tracer:

```
if s.tracer.Logger != nil {
  s.tracer.Logger("OnStart hook error: " + err.Error())
}
```